docs/reference/cli.md

```diff
@@ -1,40 +1,184 @@
 # CLI Reference
 
-`aclgen` is the CLI tool that is used translate your policy files into platform specific syntax. We use `abseil` for managing our flags and logging so in order to get a list of all flags you need to use the following command
+Aerleon contains three command line programs:
+
+* [`aclgen`](#aclgen) generates ACL files from your policy files (the primary program)
+* [`aclcheck`](#aclcheck) checks where hosts, ports and protocols are matched in a single policy file
+* [`cgrep`](#cgrep) answers queries about addresses, ports and protocols found in the definition files
+
+[`pol2yaml`](#pol2yaml) converts .pol, .inc, .svc, and .net files to equivalent YAML files. It can be found in [its own repository](https://github.com/aerleon/pol2yaml) but is documented here.
+
+### Config Files
+
+In addition to accepting command line arguments, `aclgen`, `aclcheck`, and
+`pol2yaml` will look for a config file named 'aerleon.yml' in the current directory.
+This location can be configured with the `--config_file` option. Options specified on
+the command line take precendence over options in config files.
+
+A table showing which config file options are used by which program appears below.
+
+Option | aclgen | aclcheck  | pol2yaml
+---- | --- | --- |  ---
+base_directory | ✔ | ✔ | ✔
+definitions_directory | ✔ | ✔ | ✔
+output_directory | ✔ |  | ✔
+optimize | ✔ |  |
+debug | ✔ |  |
+max_renderers | ✔ |  |
+shade_check | ✔ |  |
+exp_info | ✔ |  |
 
-```bash
-aclgen --helpfull
-```
 
-This will provide flags for each imported library that supports them. We provide below a current list of the supported flags.
+## aclgen
 
-```bash
-  --base_directory: The base directory to look for acls; typically where you'd find ./corp and ./prod
-    (default: './policies')
-  --config_file: A yaml file with the configuration options for aerleon;
+```
+  --base_directory: The base directory to search recursively for policy files.
+                    Relative policy imports are resolved against this directory.
+                    If --policy_file is used, aclgen will not search this directory.
+    Default: './policies'
+
+  --config_file: A YAML file with configuration options;
     repeat this option to specify a list of values
-  --[no]debug: Debug messages
-    (default: 'false')
-  --definitions_directory: Directory where the definitions can be found.
-    (default: './def')
-  --exp_info: Print a info message when a term is set to expire in that many weeks.
-    (default: '2')
+
+  --[no]debug: Display detailed messages. Default: 'false'
```

Comment:
Some of these have Default on a new line, others don't. Do we want to adopt a style for this with regards to documentation and the help output?

Reply:
Defaults should be on a new line always. Good catch